pkg/operator/sync_v310_00.go

```diff
@@ -0,0 +1,157 @@
+				LastGeneration: actualDeployment.ObjectMeta.Generation,
+			},
+		}
+	}
+
+	v1alpha1helpers.SetErrors(&versionAvailability, errors...)
+
+	return versionAvailability, errors
+}
+
+// TODO manage rotation in addition to initial creation
+func manageSigningSecret(c ServiceCertSignerOperator) (*corev1.Secret, bool, error) {
+	secret:= resourceread.ReadSecretV1OrDie(v310_00_assets.MustAsset("v3.10.0/service-serving-cert-signer-controller/signing-secret.yaml"))
+	existing, err := c.corev1Client.Secrets(secret.Namespace).Get(secret.Name, metav1.GetOptions{})
+	if !apierrors.IsNotFound(err){
+		return existing, false, err
+	}
+
+	ca, err := crypto.MakeCAConfig(serviceServingCertSignerName(), 10)
+	if err != nil{
+		return existing, false, err
+	}
+
+	certBytes := &bytes.Buffer{}
+	keyBytes := &bytes.Buffer{}
+	if err := ca.WriteCertConfig(certBytes, keyBytes); err != nil{
+		return existing, false, err
+	}
+
+	secret.Data["tls.crt"] = certBytes.Bytes()
+	secret.Data["tls.key"] = keyBytes.Bytes()
+
+	return resourceapply.ApplySecret(c.corev1Client, secret)
+}
+
+func ensureConfigMap_v310_00_to_00(c ServiceCertSignerOperator, options scsv1alpha1.ServiceCertSignerOperatorConfigSpec) (*corev1.ConfigMap, bool, error) {
+	// TODO use an unstructured object to merge configs
+	config, err := readServiceServingCertSignerConfig(v310_00_assets.MustAsset("v3.10.0/service-serving-cert-signer-controller/defaultconfig.yaml"))
+	if err != nil {
+		return nil, false, err
+	}
+	configBytes, err := runtime.Encode(unstructured.UnstructuredJSONScheme, config)
+	if err != nil {
+		return nil, false, err
+	}
+
+	requiredConfigMap := resourceread.ReadConfigMapV1OrDie(v310_00_assets.MustAsset("v3.10.0/service-serving-cert-signer-controller/cm.yaml"))
+	requiredConfigMap.Data["controller-config.yaml"] = string(configBytes)
+
+	return resourceapply.ApplyConfigMap(c.corev1Client, requiredConfigMap)
+}
+
+func ensureDeployment_v310_00_to_00(c ServiceCertSignerOperator, options *scsv1alpha1.ServiceCertSignerOperatorConfig, previousAvailability *operatorsv1alpha1.VersionAvailablity, forceDeployment bool) (*appsv1.Deployment, bool, error) {
+	required := resourceread.ReadDeploymentV1OrDie(v310_00_assets.MustAsset("v3.10.0/service-serving-cert-signer-controller/deployment.yaml"))
+	required.Spec.Template.Spec.Containers[0].Image = options.Spec.ImagePullSpec
+	required.Spec.Template.Spec.Containers[0].Args = append(required.Spec.Template.Spec.Containers[0].Args, fmt.Sprintf("-v=%d", options.Spec.Logging.Level))
+
+	generation := int64(-1)
+	if previousAvailability != nil {
+		for _, curr := range previousAvailability.Generations {
+			if curr.Name == "service-serving-cert-signer" {
+				generation = curr.LastGeneration
+			}
+		}
+	}
+	return resourceapply.ApplyDeployment(c.appsv1Client, required, generation, forceDeployment)
+}
+
+
+func serviceServingCertSignerName() string {
+	return fmt.Sprintf("%s@%d", "openshift-service-serving-signer", time.Now().Unix())
```

Comment:
you sure about using UNIX timestamp as name? can't we use generation or something?

Reply:
> you sure about using UNIX timestamp as name? can't we use generation or something?
> 
>  Write

I could go UID, but generation isn't unique enough.